File: tests/fixtures/lang_samples/kotlin/classes.kt
// Fixtures for class-level (LCOM4 / god-class) walker tests.
//
// Kotlin accesses instance members WITHOUT a receiver idiomatically; these
// fixtures use an explicit `this.` so the member-access node type is
// exercised. The implicit-receiver case is the documented "no signal" path.

class Cohesive {
    var total = 0
    var count = 0

    fun add(n: Int) {
        this.total += n
        this.count += 1
    }

    fun average(): Int {
        return if (this.count != 0) this.total / this.count else 0
    }

    fun reset() {
        this.total = 0
        this.count = 0
    }

    fun describe(): Int {
        return this.count
    }
}

class Splintered {
    var a = 0
    var b = 0

    fun setA(v: Int) {
        this.a = v
    }

    fun getA(): Int {
        return this.a
    }

    fun setB(v: Int) {
        this.b = v
    }

    fun getB(): Int {
        return this.b
    }

    fun loner(): Int {
        return 42
    }
}

File: tests/fixtures/lang_samples/cpp/classes.cpp
// Fixtures for class-level (LCOM4 / god-class) walker tests.
//
// Uses explicit `this->` so the member-access node type is exercised; bare
// member access (idiomatic C++) is the documented "no signal" path.

class Cohesive {
    int total;
    int count;

public:
    void add(int n) {
        this->total += n;
        this->count += 1;
    }

    int average() {
        return this->count ? this->total / this->count : 0;
    }

    void reset() {
        this->total = 0;
        this->count = 0;
    }

    int describe() {
        return this->count;
    }
};

class Splintered {
    int a;
    int b;

public:
    void setA(int v) {
        this->a = v;
    }

    int getA() {
        return this->a;
    }

    void setB(int v) {
        this->b = v;
    }

    int getB() {
        return this->b;
    }

    int loner() {
        return 42;
    }
};
